src/hooks/useCollection.js

```diff
@@ -15,48 +15,62 @@ import firestore from 'services/firebase/firestore'
 const baseSortRule = ['_createdAt', 'desc']
 
 const useCollection = (props) => {
+  const {
+    ref: propsRef,
+    where: propsWhere,
+    orderBy: propsOrderBy,
+    limit: propsLimit
+  } = props
   const [value, setValue] = useState()
-  const [loading, setLoading] = useState(true)
+  const [loading, setLoading] = useState(!!propsRef)
   const [loadingMore, setLoadingMore] = useState()
   const [error, setError] = useState()
   const [lastVisible, setLastVisible] = useState()
   const [loadMoreAvailable, setLoadMoreAvailable] = useState(true)
 
   useEffect(() => {
-    if (props?.where?.length) {
+    if (propsWhere?.length) {
       setLoadMoreAvailable(false)
     }
-  }, [props?.where])
+  }, [propsWhere])
 
   useEffect(() => {
     const fetchData = async () => {
       setLoading(true)
-      const documents = []
       try {
-        const querySnapshot = props?.where?.length
+        const querySnapshot = propsWhere?.length
           ? query(
-              collection(firestore, props?.ref),
-              ...props?.where.map((rule) => where(...rule))
+              collection(firestore, propsRef),
+              ...propsWhere.map((rule) => where(...rule)),
+              Array.isArray(propsOrderBy)
+                ? orderBy(...propsOrderBy)
+                : orderBy(...baseSortRule),
+              limit(propsLimit)
             )
           : query(
-              collection(firestore, props?.ref),
-              Array.isArray(props?.orderBy)
-                ? orderBy(...props?.orderBy)
+              collection(firestore, propsRef),
+              Array.isArray(propsOrderBy)
```

Comment:
This and 2 following rows are equal to rows 45-47, please figure out on it due to DRY principle